packages/client/src/schema/repository.ts

```diff
@@ -182,24 +189,36 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
   ): Promise<SelectedPick<Record, ['*']> | SelectedPick<Record, ['*']>[]> {
     // Create many records
     if (Array.isArray(a)) {
-      return this.#bulkInsertTableRecords(a);
+      const records = await this.#bulkInsertTableRecords(a);
+      await Promise.all(records.map((record) => this.recordCache.set(record.id, record)));
+
+      return records;
     }
 
     // Create one record with id as param
     if (isString(a) && isObject(b)) {
       if (a === '') throw new Error("The id can't be empty");
-      return this.#insertRecordWithId(a, b);
+      const record = await this.#insertRecordWithId(a, b);
+      await this.recordCache.set(record.id, record);
```

Comment:
Here and in any other place where we are storing single records, we need to generate a cache key that contains the table name and the record id, because the record id can be the same for records in different tables.